internal/versions/rest_api_versions.go

```diff
@@ -0,0 +1,210 @@
+	// validations
+	if err := validateCPUArchitecture(cpuArchitecture); err != nil {
+		return nil, err
+	}
+	versionFormat := common.GetVersionFormat(openshiftVersion)
+	if versionFormat == common.NoneVersion || versionFormat == common.MajorVersion {
+		return nil,
+			errors.Errorf(
+				"invalid openshiftVersion '%s'. Expected format: 'major.minor' or 'major.minor.patch', optionally followed by a prerelease identifier",
+				openshiftVersion,
+			)
+	}
+
+	query := h.db.Session(&gorm.Session{})
+	cleanQuery := h.db.Session(&gorm.Session{})
+	query = query.Where(&models.ReleaseImage{})
+	if cpuArchitecture == common.MultiCPUArchitecture {
+		query = query.Where("cardinality(cpu_architectures) > 1")
+	} else {
+		query = query.Where(cleanQuery.Where("? = ANY(cpu_architectures)", cpuArchitecture).Or("cpu_architecture = ?", cpuArchitecture))
+	}
+	newVersion := openshiftVersion
+
+	// Find the exact version
+	if versionFormat == common.MajorMinorPatchVersion {
+		// add multi suffix if missing
+		if cpuArchitecture == common.MultiCPUArchitecture && !strings.HasSuffix(newVersion, "-multi") {
+			newVersion = newVersion + "-multi"
+		}
+		query = query.Where("version = ?", newVersion)
+		var releaseImage models.ReleaseImage
+		err := query.Take(&releaseImage).Error
+		if err != nil {
+			if errors.Is(err, gorm.ErrRecordNotFound) {
+				return nil, errors.Errorf("no release image found for openshiftVersion: '%s' and CPU architecture '%s'", openshiftVersion, cpuArchitecture)
+			}
+
+			return nil, err
+		}
+
+		if err = assertNotIgnored(*releaseImage.Version, h.ignoredOpenshiftVersions); err != nil {
+			return nil, err
+		}
+
+		return &releaseImage, err
+	}
+
+	// openshiftVersion must be major.minor, find the latest version matching
+
+	// In case openshiftVersion is in the format of major.minor-multi
+	newVersion = strings.TrimSuffix(openshiftVersion, "-multi")
+
+	// Escape SQL wildcards in case thay are approved as a version
+	newVersion = common.EscapePlaceHolders(newVersion)
+	query = query.Where("version LIKE ?", newVersion+"."+"%")
+
+	var releaseImages models.ReleaseImages
+	err := query.Find(&releaseImages).Error
+	if err != nil {
+		return nil, err
+	}
+
+	if len(releaseImages) == 0 {
+		return nil, errors.Errorf("no release image found for openshiftVersion: '%s' and CPU architecture '%s'", openshiftVersion, cpuArchitecture)
+	}
+
+	return getLatestReleaseImage(releaseImages, h.ignoredOpenshiftVersions)
+}
+
+// GetReleaseImageByURL fetches a release image using the specified URL.
+// It returns an error if no matching image is found.
+func (h *restAPIVersionsHandler) GetReleaseImageByURL(_ context.Context, url, _ string) (*models.ReleaseImage, error) {
+	var releaseImage models.ReleaseImage
+
+	query := h.db.Session(&gorm.Session{})
+	err := query.Model(&models.ReleaseImage{}).Where("url = ?", url).Take(&releaseImage).Error
```

Comment:
I was thinking about that, but I think it I can't do much validation because urls can be in many forms (in the CI it is much different for example). I think if it will be empty it just won't find anything, do you think it worth checking it separately ?